From: Johannes Weiner <hannes@cmpxchg.org>
Subject: mm: page_alloc: speed up fallbacks in rmqueue_bulk() fix
Date: Wed, 9 Apr 2025 09:56:52 -0400

reset rmqueue_mode in rmqueue_buddy() error loop, per Yunsheng Lin

Link: https://lkml.kernel.org/r/20250409140023.GA2313@cmpxchg.org
Signed-off-by: Johannes Weiner <hannes@cmpxchg.org>
Cc: Brendan Jackman <jackmanb@google.com>
Cc: Carlos Song <carlos.song@nxp.com>
Cc: linyunsheng <linyunsheng@huawei.com>
Cc: Mel Gorman <mgorman@techsingularity.net>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/page_alloc.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/mm/page_alloc.c~mm-page_alloc-speed-up-fallbacks-in-rmqueue_bulk-fix-2
+++ a/mm/page_alloc.c
@@ -2979,7 +2979,6 @@ struct page *rmqueue_buddy(struct zone *
 {
 	struct page *page;
 	unsigned long flags;
-	enum rmqueue_mode rmqm = RMQUEUE_NORMAL;
 
 	do {
 		page = NULL;
@@ -2992,6 +2991,8 @@ struct page *rmqueue_buddy(struct zone *
 		if (alloc_flags & ALLOC_HIGHATOMIC)
 			page = __rmqueue_smallest(zone, order, MIGRATE_HIGHATOMIC);
 		if (!page) {
+			enum rmqueue_mode rmqm = RMQUEUE_NORMAL;
+
 			page = __rmqueue(zone, order, migratetype, alloc_flags, &rmqm);
 
 			/*
_
